.github/workflows/build-docker.yaml

```diff
@@ -3,41 +3,41 @@ name: Build docker container
 on: push
 
 jobs:
-    docker:
-        runs-on: ubuntu-latest
-
-        steps:
-            - name: Checkout repository
-              uses: actions/checkout@v2
-            - name: Docker meta
-              id: meta
-              uses: docker/metadata-action@v5
-              with:
-                # list of Docker images to use as base name for tags
-                images: |
-                  ericmjl/pixi-cuda-environment
-                # generate Docker tags based on the following events/attributes
-                tags: |
-                  type=sha
-                  type=raw,value=latest
-
-
-            - name: Set up QEMU
-              uses: docker/setup-qemu-action@v3
-
-            - name: Set up Docker Buildx
-              uses: docker/setup-buildx-action@v3
-
-            - name: Login to Docker Hub
-              uses: docker/login-action@v3
-              with:
-                username: ${{ secrets.DOCKERHUB_USERNAME }}
-                password: ${{ secrets.DOCKERHUB_TOKEN }}
-
-            - name: Build and push
-              uses: docker/build-push-action@v6
-              with:
-                push: true
-                tags: ${{ steps.meta.outputs.tags }}
-                cache-from: type=gha
-                cache-to: type=gha,mode=max
+  build-container:
+    name: Build container
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        # list of Docker images to use as base name for tags
+        images: |
+          ericmjl/pixi-cuda-environment
+        # generate Docker tags based on the following events/attributes
+        tags: |
+          type=sha
+          type=raw,value=latest
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
```

Comment:
Consider using GHCR as a simpler alternative to Docker Hub. No credentials are required. See [maresb/simple-ghcr-example](https://github.com/maresb/simple-ghcr-example).